### DISCUSSION TITLE
Cannot call functions in event hooks handlers

### DISCUSSION THREAD

**Discussion by bennythemink** [pocketbase/pocketbase#5240] [Q&A] [answered]
Hi everyone,

I'm having a little trouble with event hooks and hoping someone can point me in the right direction. I have a main.pb.js file where I use the '**onRecordAfterCreateRequest**' hook to send an email when a new record appears. This works well.

I'm having two problems that may be related: the first is that I want to call other functions defined in the same file in the onRecordAfterCreateRequest handler but they are not being called, even a test function defined at the top of the file that only does a console log is not working. 

The second problem is in a separate test I call $app.dao.findRecordById but it's not working. I've ensured that the collection API rules are open so they are not the problem. I suspect it's related to the first problem above. I've included the line below for reference. Thanks in advance for any help.

`const organisation = $app.dao.findRecordById('organisations', e.record.get('organisation'))`

**Comment by ganigeorgiev** [comment]
About the first issue, please check https://pocketbase.io/docs/js-overview/#handlers-scope.

For the second one, there are not enough details to understand where the above function is invoked or what is even the returned error (_you can check the logs in the Admin UI > Logs and/or start the executable with `--dev`_). Also note that db calls can be executed only after app bootstrap. If you need further help, please elaborate more what you are trying to do and paste a more detailed code sample with the failing code to understand the broader picture and the execution context.

**Reply by ganigeorgiev** [reply, marked as answer]
_Also I'm not sure if it is just a typo in the example but note that `$app.dao` is a method, aka. it needs to be `$app.dao()`._

**Reply by bennythemink** [reply]
Thank you @ganigeorgiev, you were spot on, a stupid typo on my part for the $app.dao. Thank you for pointing me to the handler scope, apologies I missed that.
